db/blob/blob_fetcher.h

```diff
@@ -9,18 +9,28 @@
 #include "rocksdb/status.h"
 
 namespace ROCKSDB_NAMESPACE {
+
 class Version;
+class Slice;
+class FilePrefetchBuffer;
+class PinnableSlice;
+class BlobIndex;
 
 class BlobFetcher {
  public:
-  BlobFetcher(Version* version, const ReadOptions& read_options)
+  BlobFetcher(const Version* version, const ReadOptions& read_options)
       : version_(version), read_options_(read_options) {}
 
-  Status FetchBlob(const Slice& user_key, const Slice& blob_index,
-                   PinnableSlice* blob_value);
+  Status FetchBlob(const Slice& user_key, const Slice& blob_index_slice,
+                   FilePrefetchBuffer* prefetch_buffer,
+                   PinnableSlice* blob_value, uint64_t* bytes_read) const;
+
+  Status FetchBlob(const Slice& user_key, const BlobIndex& blob_index,
+                   FilePrefetchBuffer* prefetch_buffer,
+                   PinnableSlice* blob_value, uint64_t* bytes_read) const;
 
  private:
-  Version* version_;
+  const Version* version_;
```

Comment:
Do we allow copy/move, assignment operator for `BlobFetcher`? I forgot the reason we didn't mark the copy constructors, etc. as `= delete`. Just want to catch up on this.

Reply:
Well I'm not the original author of `BlobFetcher` but technically it's perfectly safe to copy around.

Reply:
Just to clarify, this switch from `Version*` to `const Version*` does not affect the ability to copy-construct or assign the type like using a `const` pointer (`* const`) would.